src/daemon/main.c

```diff
@@ -1204,6 +1204,17 @@ static void get_netdata_configured_variables()
 
     // ------------------------------------------------------------------------
     // get default database size
+    dbengine_tier0_in_ram = appconfig_get_boolean(
+        &netdata_config,
+        CONFIG_SECTION_DB,
+        "dbengine tier 0 in ram",
+        CONFIG_BOOLEAN_NO);
+
+    if (dbengine_tier0_in_ram && default_rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE) {
+        netdata_log_info("Tier 0 was requested to be in RAM, changing agent default to RAM");
+        default_rrd_memory_mode = RRD_MEMORY_MODE_RAM;
+        config_set(CONFIG_SECTION_DB, "db", rrd_memory_mode_name(default_rrd_memory_mode));
```

Comment:
> Tier 0 was requested to be in RAM, changing agent default to RAM

Tier 0 should be in RAM only, no? Mode should remain dbengine for tiers 1+.

Reply:
Yes, tiers 1+ can only be dbengine.

Reply:
ok, but why does it say "changing agent default to RAM"?

Reply:
It overrides the `db` setting to RAM (netdata.conf will show ram)

Reply:
I think it shouldn't. The actual configuration is 

```
mode = dbengine
dbengine tier 0 in ram = yes
```

Reply:
```
dbengine tier 0 in ram = yes
```

will now **only** initialize the localhost to memory mode RAM without touching the default memory mode